### PR TITLE
add github actions to build jars & docker images

### PR DESCRIPTION
**Pull request by topi314**
to whitelist github actions:
```
actions/checkout@v3
actions/setup-java@v3
gradle/gradle-build-action@v2
actions/upload-artifact@v3
actions/download-artifact@v2
docker/metadata-action@v4
docker/setup-qemu-action@v2
docker/setup-buildx-action@v2
docker/build-push-action@v3
softprops/action-gh-release@v1
```

to configure secrets:
```
DOCKERHUB_USERNAME
DOCKERHUB_TOKEN
```

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      - name: Execute Gradle build
+        run: ./gradlew build
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: Lavalink.jar
+          path: LavalinkServer/build/libs/Lavalink.jar
+
+  docker:
```
Comment (freyacodes) [inline .github/workflows/build.yml:36]: Could you make this skip if there is no credentials available? It seems to be failing instead

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
+name: Build
+
+on:
+  push:
+    paths-ignore:
+      - '**.md'
+  pull_request:
```
Comment (freyacodes) [inline .github/workflows/build.yml:7]: I think this line is unnecessary. Pull requests trigger the push event, making it redundant in this case. By removing this you can simplify the run conditions and remove many of the "skipped" commit statuses

**Review by freyacodes — CHANGES_REQUESTED** [review]

**Review by freyacodes — APPROVED** [review]
I'm actually going to merge this now such that we can actually benefit from it. Please continue development in a new PR

File: .github/workflows/build.yml (hunk 1 above)
Comment (topi314) [reply .github/workflows/build.yml:36]: resolved in https://github.com/freyacodes/Lavalink/pull/698

File: .github/workflows/build.yml (hunk 2 above)
Comment (topi314) [reply .github/workflows/build.yml:7]: resolved in https://github.com/freyacodes/Lavalink/pull/698

**Pull request merged**